rcl_logging_log4cxx/src/rcl_logging_log4cxx/rcl_logging_log4cxx.cpp

```diff
@@ -123,8 +117,40 @@ rcl_logging_ret_t rcl_logging_external_initialize(const char * config_file, rcut
     // Set the default File Appender on the root logger
     log4cxx::LoggerPtr root_logger(get_logger(nullptr));
     log4cxx::LayoutPtr layout(new log4cxx::PatternLayout(LOG4CXX_STR("%m%n")));
-    char log_name_buffer[128] = {0};
-    snprintf(log_name_buffer, sizeof(log_name_buffer), DEFAULT_LOG_FILE, GET_PID());
+
+    // To be compatible with ROS 1, we want to construct a default filename of
+    // the form ~/.ros/log/<exe>_<pid>_<milliseconds-since-epoch>.log
+
+    // First get the home directory.
+    const char *homedir = rcutils_get_home_dir();
+    if (homedir == NULL) {
+      // We couldn't get the home directory; it is not really going to be
+      // possible to do logging properly, so get out of here without setting
+      // up logging.
+      return RC_LOGGING_RET_ERROR;
+    }
+
+    // Now get the milliseconds since the epoch in the local timezone.
+    rcutils_time_point_value_t now;
+    rcutils_ret_t ret = rcutils_system_time_now(&now);
+    if (ret != RCUTILS_RET_OK) {
+      // We couldn't get the system time, so get out of here without setting up
+      // logging.
+      return RC_LOGGING_RET_ERROR;
+    }
+    int64_t ms_since_epoch = RCUTILS_NS_TO_MS(now);
+
+    // Get the program name.
+    char *basec = rcutils_get_executable_name(allocator);
+    if (basec == NULL) {
+      // We couldn't get the program name, so get out of here without setting up
+      // logging.
+      return RC_LOGGING_RET_ERROR;
+    }
+
+    char log_name_buffer[512] = {0};
+    snprintf(log_name_buffer, sizeof(log_name_buffer), "%s/.ros/log/%s_%i_%" PRId64 ".log", homedir, basec, rcutils_get_pid(), ms_since_epoch);
```

Comment:
This line looks really long, is there any linting happening in this file?

Reply:
That's true.  The linter isn't enabled on this package.  I'll open an issue to track turning on the linter and dealing with the fallout.